inference-engine/tests/functional/plugin/shared/src/subgraph_tests/matmul_squeeze_add.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <ie_core.hpp>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <vector>
+#include <ie_plugin_config.hpp>
+
+#include "common_test_utils/common_utils.hpp"
+#include "functional_test_utils/blob_utils.hpp"
+#include "functional_test_utils/layer_test_utils.hpp"
+#include "functional_test_utils/plugin_cache.hpp"
+#include "ngraph_functions/pass/convert_prc.hpp"
+
+#include "subgraph_tests/matmul_squeeze_add.hpp"
+
+#include "ngraph_functions/builders.hpp"
+
+namespace LayerTestsDefinitions {
+
+std::string MatmulSqueezeAddTest::getTestCaseName(testing::TestParamInfo<matmulSqueezeAddParams> obj) {
+    InferenceEngine::Precision netPrecision;
+    InferenceEngine::SizeVector inputShapes, newInputShapes;
+    std::string targetDevice;
+    std::map<std::string, std::string> configuration;
+    std::tie(netPrecision, targetDevice, configuration) = obj.param;
+
+    std::ostringstream result;
+    result << "IS=" << CommonTestUtils::vec2str(inputShapes) << "_";
+    result << "netPRC=" << netPrecision.name() << "_";
+    result << "targetDevice=" << targetDevice;
+    for (auto const& configItem : configuration) {
+        result << "_configItem=" << configItem.first << "_" << configItem.second;
+    }
+    return result.str();
+}
+
+void MatmulSqueezeAddTest::SetUp() {
+    InferenceEngine::Precision netPrecision;
+    std::tie(netPrecision, targetDevice, configuration) = this->GetParam();
+    auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+
+    auto params = ngraph::builder::makeParams(ngPrc, { {1, 512} });
+    outPrc = InferenceEngine::Precision::FP32;
+
+    auto constant_0 = ngraph::builder::makeConstant<float>(ngPrc, { 1000, 512 }, {}, true);
```

Comment:
this method is flawed it generates only integer_like values, better create any vector with real float values and use it in construction of constants. It will help to check issues with quantizations, since values from this generator will work even if scale_factor = 1 is used for them which might be error in code and won't be caught. You could also modify inputs so they are no int-like but modifying weights is easier.